homeassistant/components/switcher_kis/__init__.py

```diff
@@ -58,23 +98,63 @@
         """On homeassistant stop, gracefully stop the bridge if running."""
         await v2bridge.stop()
 
-    hass.async_add_job(hass.bus.async_listen_once(
-        EVENT_HOMEASSISTANT_STOP, async_stop_bridge))
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_stop_bridge)
 
     try:
         device_data = await wait_for(
             v2bridge.queue.get(), timeout=5.0, loop=hass.loop)
     except (Asyncio_TimeoutError, RuntimeError):
-        _LOGGER.exception("failed to get response from device")
+        _LOGGER.exception("Failed to get response from device")
         await v2bridge.stop()
         return False
 
     hass.data[DOMAIN] = {
         DATA_DEVICE: device_data
     }
 
+    async def async_switch_platform_discovered(
+            platform: str, discovery_info: Optional[Dict]) -> None:
+        """Use for registering services after switch platform is discoverd."""
+        if platform != DOMAIN:
+            return
+
+        async def async_set_auto_off_service(service: ServiceCallType) -> None:
+            """Use for handling setting device auto-off service calls."""
+            from aioswitcher.api import SwitcherV2Api
+
+            await _validate_edit_permission(
+                hass, service.context, service.data[CONF_ENTITY_ID])
+
+            async with SwitcherV2Api(hass.loop, device_data.ip_addr, phone_id,
+                                     device_id, device_password) as swapi:
+                await swapi.set_auto_shutdown(service.data[CONF_AUTO_OFF])
+
+        hass.services.async_register(
+            DOMAIN, SERVICE_SET_AUTO_OFF_NAME,
+            async_set_auto_off_service,
+            schema=SERVICE_SET_AUTO_OFF_SCHEMA)
+
+        async def async_update_name_service(service: ServiceCallType) -> None:
+            """Use for handling update device name service calls."""
+            from aioswitcher.api import SwitcherV2Api
+
+            await _validate_edit_permission(
+                hass, service.context, service.data[CONF_ENTITY_ID])
+
+            async with SwitcherV2Api(hass.loop, device_data.ip_addr, phone_id,
+                                     device_id, device_password) as swapi:
+                await swapi.set_device_name(service.data[CONF_NAME])
+
+        hass.services.async_register(
+            DOMAIN, SERVICE_UPDATE_DEVICE_NAME_NAME,
```

Comment:
What's the use of allowing people to set a device name? Since it has a unique ID, users can just use the entity registry to change the name and entity ID. 

We should only offer things as a service that might be used for automations. Administrative tasks should be done in the app for the product.

Reply:
@balloob 
Yeah but that entity registry modification will of course only reflect in home assistant.
This service sets the actual device name so it's also reflected in the vendor's app.

It's on a nice-to-have basis I believe,
I can take or leave it.

Please advice...

Reply:
Let's drop it. We generally avoid putting device management into HA (exceptions if integration is about a protocol like Zigbee/Z-Wave)

Reply:
@balloob 
Removed in [this commit](https://github.com/home-assistant/home-assistant/pull/23477/commits/61ec2693956dd86bbf9b9f1c58c3c99a4442ee06).

Thanks!
:-)